bin/convert-formula.sh

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env sh
+set -o nounset # Treat unset variables as an error and immediately exit
+set -o errexit # If a command fails exit the whole script
+
+if [ "${DEBUG:-false}" = "true" ]; then
+  set -x # Run the entire script in debug mode
+fi
+
+usage() {
+  echo "usage: $(basename "$0") <new-formula-name>" 1>&2
+  echo 1>&2
+  echo "Convert template-formula to <new-formula-name>-formula." 1>&2
+  echo "<new-formula-name> should be a string of lowercase characters and numbers only." 1>&2
+  echo "<new-formula-name> should not be any of 'bin' 'docs' 'test'." 1>&2
+}
+
+args() {
+  if [ $# -ne 1 ]; then
+    usage
+    exit 1
+  fi
+  NEW_NAME=$1
+  if echo "$NEW_NAME" | grep -E --quiet --invert-match '^[a-z0-9]+$'; then
+    usage
+    exit 1
+  fi
+  if echo bin docs test | grep --quiet --word-regexp "$NEW_NAME"; then
+    usage
+    exit 1
+  fi
+}
+
+convert_formula() {
+  # Empty history and make commit message `semantic-release`-compliant
+  # Works for forks of `template-formula` as well as GitHub "Use this template"
+  # See https://stackoverflow.com/a/15572071/5009408
+  git reset \
+    "$(echo 'feat: initial commit' \
+      | git commit-tree 'HEAD^{tree}')"
+  git rm --quiet bin/convert-formula.sh AUTHORS.md CHANGELOG.md
+  git mv TEMPLATE "${NEW_NAME}"
+  grep --recursive --files-with-matches --exclude-dir=.git TEMPLATE . \
+    | xargs -L 1 ex -sc '%s/TEMPLATE/'"${NEW_NAME}"'/g|x'
```

Comment:
In my first local tests, the script doesn't get past this line.  Since this is working in Travis, there's probably something in the way I've patched things together.  The only way past it is to comment out line 3, i.e.:

```sh
# set -o errexit # If a command fails exit the whole script
```

Reply:
You might get some clues by running as (if you haven't already done so):
```DEBUG=true bin/convert-formula.sh ...```

Reply:
Yes, I started with that and then tried all sorts, including removing all of the `quiet` or `silent` flags but to no avail.  Even when I get it running with this line commented out, there are no warning given.  Not sure what's happening with that `ex` command to cause an error status...

Reply:
@dafyddj This sounds a little strange but I found a solution (using `-u NONE`) based on this reference:

* https://unix.stackexchange.com/questions/14497/why-would-vim-return-non-zero-exit-code-if-i-exit-immediately-after-opening

So that's:

```sh
  grep --recursive --files-with-matches --exclude-dir=.git TEMPLATE . \
    | xargs -L 1 ex -u NONE -sc '%s/TEMPLATE/'"${NEW_NAME}"'/g|x'
```

* This actually worked, with `set -o errexit` re-enabled.
* Maybe the end user (like me) could have something strange in their configuration, affecting `ex` when it exits?